src/ansi.rs

```diff
@@ -0,0 +1,1607 @@
+//! ANSI Terminal Stream Parsing.
+
+extern crate alloc;
+use alloc::{borrow::ToOwned, string::String, vec::Vec};
```

Comment:
This would basically make this crate STD. We should try to make things allocation free.

Reply:
Looks like it's mainly the window title handling code. I'll look at it over the next week or so

Reply:
I think it's okay to do a little less in some cases, usually handling collection from bytes into Strings is not that big of a deal.

So I think we should be able to completely remove allocations.

Reply:
I've done this and I believe the only external change is the signature of `set_title`:
```rust
fn set_title(&mut self, _: Option<&str>)
```
Previously: `Option<String>`

Using ArrayVec, ArrayString -- allocation sizes are `MAX_PARAMS` or `MAX_OSC_RAW` from crate root as appropriate.